pkg/datapath/loader/loader.go

```diff
@@ -790,16 +777,31 @@ func replaceWireguardDatapath(ctx context.Context, logger *slog.Logger, lnc *dat
 			)
 		}
 	}
-	// Selectively detach cil_from_netdev from ingress.
-	if !option.Config.NeedBPFHostOnWireGuardDevice() {
-		if err := detachSKBProgram(logger, device, symbolFromHostNetdevEp,
+	// Attach/detach cil_from_wireguard to/from ingress.
+	if option.Config.NeedIngressOnWireGuardDevice() {
+		if err := attachSKBProgram(logger, device, obj.FromWireguard, symbolFromWireguard,
+			linkDir, netlink.HANDLE_MIN_INGRESS, option.Config.EnableTCX); err != nil {
+			return fmt.Errorf("interface %s ingress: %w", device, err)
+		}
+	} else {
+		if err := detachSKBProgram(logger, device, symbolFromWireguard,
 			linkDir, netlink.HANDLE_MIN_INGRESS); err != nil {
 			logger.Error("",
 				logfields.Error, err,
 				logfields.Device, wgTypes.IfaceName,
 			)
 		}
 	}
+	// Cleanup previous cil_from_netdev from v1.17.
```

Comment:
So the implication is we can remove this post 1.19? Maybe add a reminder?

Reply:
Yes, in v1.19 (or even v1.18.1) we wouldn't need this anymore, since we've supposedly already taken care of that.

Do you mean a reminder in the code? (if yes, I'd wait in case of other needed changes to restart the CI process)

Reply:
Updated comment in the code.